File: sagrada_app/lib/rules/empty_dice_rule.dart
import 'package:sagrada_app/entities/dice_board.dart';
import 'package:sagrada_app/rules/rule.dart';

class EmptyDiceRule extends Rule {
  EmptyDiceRule()
      : super(
          title: 'Brak kostek',
          description: 'Punkty ujemne za każdą pustą kostkę na planszy.',
        );

  @override
  int calculateScore(DiceBoard board) {
    int emptyDiceCount = 0;

    for (int row = 0; row < DiceBoard.rows; row++) {
      for (int column = 0; column < DiceBoard.columns; column++) {
        if (board.get(row, column) == null) {
          emptyDiceCount++;
        }
      }
    }

    return -emptyDiceCount;
  }
}

File: rules/empty_dice_rule.js
class EmptyDiceRule extends Rule {
    constructor() {
        super('Puste pola', 'Każda pusta kostka w całym witrażu -1 punkt');
    }

    calculateScore(board) {
        let emptyCount = 0;

        for (let row = 0; row < DiceBoard.rows; row++) {
            for (let column = 0; column < DiceBoard.columns; column++) {
                const dice = board.get(row, column);
                if (dice === null) {
                    emptyCount++;
                }
            }
        }

        return -emptyCount;
    }
}